### PR TITLE
[GIE Compiler] Support Case When Expression in Logical and Physical Plan

### PR DESCRIPTION
**Pull request by shirly121**
## What do these changes do?
as titled.

## Related issue number

#2686

### CODE REVIEW

File: interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rex/operator/GraphCaseOperator.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.sql.*;
+import org.apache.calcite.sql.type.SqlOperandCountRanges;
+import org.apache.calcite.sql.type.SqlOperandTypeInference;
+import org.apache.calcite.sql.type.SqlTypeUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GraphCaseOperator extends SqlOperator {
```
Comment (longbinlai) [inline interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rex/operator/GraphCaseOperator.java:37]: The case operator, should not be `GraphCaseOperator`, as it is not a `Graph` operator.

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/CaseTest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import org.apache.calcite.rex.RexCall;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class CaseTest {
+    // case when a.name = 'marko' then 1 else 3 end
+    @Test
+    public void case_1_test() {
```
Comment (longbinlai) [inline interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/CaseTest.java:35]: Add test of more than two branches

File: interactive_engine/executor/ir/proto/expr.proto [hunk 3]
```diff
@@ -124,6 +124,15 @@ message DynamicParam {
   common.IrDataType data_type = 3;
 }
 
+message Case {
```
Comment (longbinlai) [inline interactive_engine/executor/ir/proto/expr.proto:127]:
```
message Case {
  // Pair of when <expr> then <expr>
  message WhenThen {
    Expression when = 1;
    Expression then = 2;
  }
  // A case operator always follows with a sequence of when_then pairs, e.g.
  // CASE WHEN a.name == 'Marko' THEN 1
  //          WHEN a.name == 'John' THEN 2
  repeated WhenThen when_thens = 1;
  // The trailed else expression:  ELSE 3
  Expression else_result = 2;
}
```

File: interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rex/operator/GraphCaseOperator.java (hunk 1 above)
Comment (shirly121) [reply interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rex/operator/GraphCaseOperator.java]: done

File: interactive_engine/executor/ir/proto/expr.proto (hunk 3 above)
Comment (shirly121) [reply interactive_engine/executor/ir/proto/expr.proto:127]: done

**Review by longbinlai — APPROVED** [review]

**Review by zhanglei1949 — APPROVED** [review]
LGTM

**Pull request merged**